cargo-pgrx/src/command/upgrade.rs

```diff
@@ -279,27 +279,50 @@ impl CommandExecute for Upgrade {
                 // Name of the package matches the name of the root workspace 
                 // crate, do not dig for a contained package.
                 (Some(true), _) => {
-                    self.process_manifest(&path, &mut manifest)?;
+                    self.process_manifest(&path, &mut manifest)
                 },
                 // Contained package specified and this is a workspace,
                 // OR workspace has no root package name.
                 // find the correct manifest for the child crate.
                 (Some(false), Some(work))
                 | (None, Some(work)) => {
-                    println!("{:#?}", work.members);
-                    todo!()
+                    // Unfortunately cargo_toml::Workspace provides the member
+                    // list as directories but not as package names, so we have
+                    // to do a little bit of finagling here. 
+                    let mut child_path_maybe = None;
+                    for member in &work.members {
+                        if member == &package { 
+                            // Canonicalized, should have a parent path
+                            let root_path = path.parent()
+                                .ok_or(eyre!("Failed to canonicalize path, no \
+                                parent directory found."))?;
+                            let member_path = root_path
+                                .join(PathBuf::from(member))
```

Comment:
Converting `member` into a `PathBuf` here shouldn't be necessary because the arg type of `Path::join` is `impl AsRef<Path>`, and this is a String. Right?